### PR TITLE
[IMP] software_reseller: use boolean field for licenses

### PR DESCRIPTION
**Pull request by dhrs-odoo**
In this PR we have added a boolean field so that at time of project creation user can specify that if he want to have it as license or not.

Task-4273451

### CODE REVIEW

File: software_reseller/data/ir_actions_act_window.xml [hunk 1]
```diff
@@ -12,7 +12,7 @@
         <field name="view_id" ref="project.view_project_kanban" />
         <field name="view_mode">kanban,list,form</field>
         <field name="search_view_id" ref="project.view_project_project_filter" />
-        <field name="domain">[('name', 'ilike', '%licenses%')]</field>
+        <field name="domain">[('x_is_licence', '=', 'True')]</field>
```
Comment (vava-odoo) [inline software_reseller/data/ir_actions_act_window.xml:15]: We use American English in Odoo, could you fix everywhere?
```suggestion
        <field name="domain">[('x_is_license', '=', 'True')]</field>
```

File: software_reseller/data/ir_ui_view.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+                <group name="group_licence_management">
+                    <group name="group_is_licence" string="Licence" class="row mt16 o_settings_container col-lg-6">
+                        <setting class="col-lg-12" help="Make the project visible in the Licences app" id="x_is_licence_id">
+                            <field name="x_is_licence" />
+                        </setting>
+                    </group>
+                </group>
+            </xpath>
+        </field>
+      </record>
```
Comment (vava-odoo) [inline software_reseller/data/ir_ui_view.xml:22]:
```suggestion
    </record>       
```

File: software_reseller/data/ir_ui_view.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    <record id="project_project_form_customization" model="ir.ui.view">
+        <field name="name">project.project.form customization</field>
+        <field name="model">project.project</field>
+        <field name="inherit_id" ref="project.edit_project"/>
+        <field name="mode">extension</field>
+        <field name="priority">240</field>
+        <field name="type">form</field>
+        <field name="active" eval="True"/>
+        <field name="arch" type="xml">
+            <xpath expr="//page[@name='settings']/group[2]" position="inside">
```
Comment (vava-odoo) [inline software_reseller/data/ir_ui_view.xml:12]: More stable?
```suggestion
            <xpath expr="//page[@name='settings']//group[@name='group_documents_analytics']" position="inside">
```

Comment (dhrs-odoo) [reply software_reseller/data/ir_ui_view.xml]: It is not working 
image

**Comment by dhrs-odoo** [comment]
Hello @vava-odoo 
All the suggested changes has been completed 
Kindly have a look at it once you are free
Thank You :innocent:

**Review by vava-odoo — APPROVED** [review]

**Pull request merged**